code/game/objects/items/toys.dm

```diff
@@ -547,70 +600,510 @@
+		losses++	
+		playsound(src, 'sound/effects/light_flicker.ogg', 20, TRUE)
+		src_controller.visible_message("<span class='notice'> [src] collapses!</span>", \
+						"<span class='notice'> [src] collapses!</span>", null, COMBAT_MESSAGE_RANGE)
+		if(!attacker.quiet)
+			attacker.say("[pick(winlines)]")
+		attacker_controller.visible_message("<span class='notice'> [attacker] demolishes [src] and walks away victorious!</span>", \
+							"<span class='notice'> You raise up [attacker] proudly over [src]</span>!")
+	else //both win?
+		if(!quiet)
+			say("I WENT EASY ON YOU.")
+		if(!attacker.quiet)
+			attacker.say("OF COURSE YOU DID.")
+
+	in_combat = FALSE
+	attacker.in_combat = FALSE
+
+	combat_health = max_combat_health									
+	attacker.combat_health = attacker.max_combat_health
+
+	return
+
+/**
+  * This proc checks if a battle can be initiated between src and attacker.
+  *
+  * Both SRC and attacker (if attacker is included) timers are checked if they're on cooldown, and 
+  * both SRC and attacker (if attacker is included) are checked if they are in combat already.
+  * If any of the above are true, the proc returns FALSE and sends a message to user (and target, if included) otherwise, it returns TRUE
+  * 
+  * Arguments
+  *															
+  *		user:		the user who is initiating the battle
+  *		attacker:	optional arg for checking two mechs at once 
+  *		target:		optional arg used in Mech PvP battles (if used, attacker is target's toy)
+  */
+/obj/item/toy/prize/proc/check_battle_start(mob/living/carbon/user, obj/item/toy/prize/attacker, mob/living/carbon/target) 
+	if(attacker && attacker.in_combat)
+		to_chat(user, "<span class='notice'>[target?target.p_their() : "Your" ] [attacker.name] is in combat.</span>")
+		target?.to_chat(target, "<span class='notice'>Your [attacker.name] is in combat.</span>")
+		return FALSE 
+	if(in_combat)
+		to_chat(user, "<span class='notice'>Your [name] is in combat.</span>")
+		target?.to_chat(target, "<span class='notice'>[target.p_their()] [name] is in combat.</span>")
+		return FALSE 
+	if(attacker && attacker.timer > world.time)
+		to_chat(user, "<span class='notice'>[target?target.p_their() : "Your" ] [attacker.name] isn't ready for battle.</span>")
+		target?.to_chat(target, "<span class='notice'>Your [attacker.name] isn't ready for battle.</span>")
+		return FALSE 
+	if(timer > world.time)
+		to_chat(user, "<span class='notice'>Your [name] isn't ready for battle.</span>")
+		target?.to_chat(target, "<span class='notice'>[target.p_their()] [name] isn't ready for battle.</span>")
+		return FALSE 
+
+	return TRUE
+
+/**
+  * Processes any special attack moves that happen in the battle (called in the mechaBattle proc). 
+  *
+  * Makes the toy shout their special attack cry and updates its cooldown. Then, does the special attack.
+  * Arguments
+  *															
+  *		src:		the toy, doing its special move(not an arg but relevant) 	
+  *		victim:		the toy being hit by the special move
+  */
```

Comment:
It's not particularly necessary to specify what the src will be in most cases. If you really do want to keep it in, have it be in the main description of the function rather than the arguments section